bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -236,6 +234,27 @@ namespace bftEngine
 			// and the process_message() method has not start working yet
 			SimpleAutoResetEvent startSyncEvent;
 
+                        //******** METRICS ************************************
+                        concordMetrics::Component metricsComponent_;
+
+                        typedef concordMetrics::Component::Handle<
+                          concordMetrics::Gauge> GaugeHandle;
+                        typedef concordMetrics::Component::Handle<
+                          concordMetrics::Status> StatusHandle;
+                        typedef concordMetrics::Component::Handle<
+                          concordMetrics::Counter> CounterHandle;
+
+                        GaugeHandle metric_view_;
+                        GaugeHandle metric_last_stable_seq_num__;
+                        GaugeHandle metric_last_executed_seq_num_;
+                        GaugeHandle metric_last_agreed_view_;
+
+                        CounterHandle metric_slow_path_count_;
+                        CounterHandle metric_received_msgs_;
```

Comment:
We probably need to split metric_received_msgs_ to different message kinds (i.e. reads/writes).

Reply:
I changed this to include a separate counter for each external message.